src/pushNotifications/PermissionHandling.ts

```diff
@@ -0,0 +1,108 @@
+import * as Permissions from 'expo-permissions';
+import * as Notifications from 'expo-notifications';
+import Constants from 'expo-constants';
+import { Alert, Platform } from 'react-native';
+import { addToStore, readFromStore } from '../helpers';
+import { PermissionStatus } from 'expo-permissions';
+import { texts } from '../config';
+import { handleIncomingToken, PushNotificationStorageKeys } from './TokenHandling';
+
+const setInAppPermission = async (newValue: boolean) => {
+  const oldValue = await readFromStore(PushNotificationStorageKeys.IN_APP_PERMISSION);
+  if (newValue !== oldValue) {
+
+    // FIXME error handling: sync with server failed, etc
+    addToStore(PushNotificationStorageKeys.IN_APP_PERMISSION, newValue);
+
+    if (newValue) {
+      const hasPermission = await handleSystemPermissions();
+
+      if (!hasPermission) {
+        showSystemPermissionMissingDialog();
+      } else {
+        registerForPushNotificationsAsync()
+          .then(handleIncomingToken);
+      }
+    } else {
+      // remove token from store and notify server
+      handleIncomingToken();
+    }
+  }
+};
+
+export const initializePushPermissions = async () => {
+  const inAppPermission = await readFromStore(PushNotificationStorageKeys.IN_APP_PERMISSION);
+  inAppPermission && updatePushToken();
+
+  // this will only show the alert if inAppPermission is undefined (or null), but not if it is false
+  inAppPermission ?? showInitialPushAlert();
+};
+
+const registerForPushNotificationsAsync = async (): Promise<string | undefined> => {
+  const { data: token } = await Notifications.getExpoPushTokenAsync();
+
+  if (Platform.OS === 'android') {
```

Comment:
we are using a `device` object from `configs` with `device.platform` instead of `Platform` directly, to avoid the additional `Platform` import in components.
as you already import `texts` from `../configs`, you can add `device` there and use it here.